src/libs/actions/OnyxUpdateManager.ts

```diff
@@ -22,37 +21,38 @@ import * as OnyxUpdates from './OnyxUpdates';
 // The circular dependency happens because this file calls API.GetMissingOnyxUpdates() which uses the SaveResponseInOnyx.js file
 // (as a middleware). Therefore, SaveResponseInOnyx.js can't import and use this file directly.
 
-let lastUpdateIDAppliedToClient = 0;
+let lastUpdateIDAppliedToClient: number | null = 0;
 Onyx.connect({
     key: ONYXKEYS.ONYX_UPDATES_LAST_UPDATE_ID_APPLIED_TO_CLIENT,
-    callback: (val) => (lastUpdateIDAppliedToClient = val),
+    callback: (value) => (lastUpdateIDAppliedToClient = value),
 });
 
 export default () => {
     console.debug('[OnyxUpdateManager] Listening for updates from the server');
     Onyx.connect({
         key: ONYXKEYS.ONYX_UPDATES_FROM_SERVER,
-        callback: (val) => {
-            if (!val) {
+        callback: (value) => {
+            if (!value) {
                 return;
             }
 
             // Since we used the same key that used to store another object, let's confirm that the current object is
             // following the new format before we proceed. If it isn't, then let's clear the object in Onyx.
             if (
-                !_.isObject(val) ||
-                !_.has(val, 'type') ||
-                (!(val.type === CONST.ONYX_UPDATE_TYPES.HTTPS && _.has(val, 'request') && _.has(val, 'response')) && !(val.type === CONST.ONYX_UPDATE_TYPES.PUSHER && _.has(val, 'updates')))
+                value === null ||
```

Comment:
Using `value === null` is useless at this point because you are already checking `value` few lines above, and is not a substitute for `isObject`. We can use `!(typeof value === 'object' && !!value)` ([source](https://underscorejs.org/docs/modules/isObject.html)) to check if `value` is not a object.

Also, I think we can simplify this condition. WDYT?

```ts
            if (
                !(typeof value === 'object' && !!value) ||
                !Object.hasOwn(value, 'type') ||
                (!(value.type === CONST.ONYX_UPDATE_TYPES.HTTPS && value.request && value.response) && !(value.type === CONST.ONYX_UPDATE_TYPES.PUSHER && value.updates))
            ) {
```